site/content/en/docs/Guides/feature-stages.md

```diff
@@ -0,0 +1,121 @@
+  applications that rely on the feature.
+* Recommended for only non-business-critical uses because of potential for incompatible changes in subsequent releases.
+  If you have multiple clusters that can be upgraded independently, you may be able to relax this restriction.
+
+> Note: Please do try `Beta` features and give feedback on them! After they exit beta, it may not be practical for us
+> to make more changes.
+
+### Stable
+
+A `Stable` feature means:
+
+* The feature is enabled and the corresponding feature gate no longer exists.
+* Stable versions of features will appear in released software for many subsequent versions.
+
+## Feature Stage Indicators
+
+There are a variety of features with Agones, how can we determine what stage each feature is in?
+
+Below are indicators for each type of functionality that can be used to determine the feature stage for a given aspect
+of Agones.
+
+### Custom Resource Definitions (CRDs)
+
+This refers to Kubernetes resource for Agones, such as `GameServer`, `Fleet` and `GameServerAllocation`.
+
+#### New CRDs
+
+For new resources, the stage of the resource will be indicated by the `apiVersion` of the resource.
+
+For example: `apiVersion: "agones.dev/v1"` is a `stable` resource, `apiVersion: "agones.dev/v1beta1"` is a `beta`
+ stage resource, and `apiVersion: "agones.dev/v1alpha1"` is a `alpha` stage resource.
+
+#### New CRD attributes
+
+`alpha` and `beta` attributes will have a corresponding `alpha` or `beta` parent element in their configuration to
+delineate their feature stage.
+
+If no such parent exists, this attribute is a `stable` feature.
+
+For example, if we were to add a hypothetical `alpha` configuration option of `timeoutSeconds` to the `GameServer`
+ resource, it would be configured like so:
+ 
+```yaml
+apiVersion: "agones.dev/v1"
+kind: GameServer
+metadata:
+  generateName: "simple-udp-"
+spec:
+  alpha: # this is the alpha feature block
+    timeoutSeconds: 30
+  ports:
+  - name: default
+    portPolicy: Dynamic
+    containerPort: 7654
+  template:
+    spec:
+      containers:
+      - name: simple-udp
+        image: gcr.io/agones-images/udp-server:0.15
+``` 
+
+As resource attributes progress through their stages, there may be breaking changes, as backward conversion between
+ `alpha`, `beta` and `stable` positioning in the resource configuration may not be guaranteed.
+
+### Agones Game Server SDK
+
+Any `alpha` or `beta` Game Server SDK functionality will have a comment block before each function declaring it either
+`alpha` or `beta`, as well as documentation indicating it as such.
+
+Any other SDK functionality not marked as `alpha` or `beta` can be assumed to be `stable`.
+
+### REST & gRPC APIs 
+
+REST and gRPC API will have versioned paths were appropriate to indicate their feature stage.
+
+For example, a REST API with a prefix of `v1alpha1` is an `alpha` stage feature.
```

Comment:
Can you please clarify?

Reply:
Should probably have examples?

Basically I'm saying if the REST Apis's path is something like: `/v1alpha1/dothing/` then it's alpha. Does that make sense?

Reply:
Yes, it would be useful to make an example or provide a URI versioning template e.g. <api-version>/<noun>. https://restfulapi.net/versioning/
Also, we should be more specific about proto versioning. Proto package and directory should reflect the version.

Reply:
Added a REST example, WDYT - is that enough?